### PR TITLE
updates slips for week4 wdi8

### PR DESCRIPTION
**Pull request by nolds9**

### CODE REVIEW

File: slips.json [hunk 1]
```diff
@@ -38,10 +38,33 @@ var data_slips = {
   ],
   "Week 04": [
     "Describe the difference between a fork and a branch.",
-    "Describe the difference between what is truthy/falsey in Ruby vs JavaScript?",
-    "Define what a function is in Javascript and how it compares to a Ruby method.",
     "Differentiate between function declarations and expressions.",
     "Explain the concept of a 'callback' function.",
+    "What's the difference between an index and a key?",
```
Comment (mattscilipoti) [inline slips.json:43]: We haven't started databases yet.

File: slips.json [hunk 2]
```diff
@@ -38,10 +38,33 @@ var data_slips = {
...
+    "Explain the purpose of HTML attributes.",
+    "What's the difference between an implicit return and a side effect?",
+    "Why would we want to take an object-oriented approach to programming?",
+    "List common steps to follow when solving a problem.",
+    "What's the difference between 'relative' and 'absolute' positioning?",
+    "What is a reliable source for 3rd party fonts?",
+    "Name 2 pseudo-classes that allow css to add content to a page.",
```
Comment (mattscilipoti) [inline slips.json]: Not in https://github.com/ga-wdi-lessons/css-review

File: slips.json [hunk 3]
```diff
@@ -38,10 +38,33 @@ var data_slips = {
...
-    "Define what a function is in Javascript and how it compares to a Ruby method.",
     "Differentiate between function declarations and expressions.",
     "Explain the concept of a 'callback' function.",
+    "What's the difference between an index and a key?",
+    "Explain the purpose of HTML attributes.",
+    "What's the difference between an implicit return and a side effect?",
+    "Why would we want to take an object-oriented approach to programming?",
+    "List common steps to follow when solving a problem.",
+    "What's the difference between 'relative' and 'absolute' positioning?",
+    "What is a reliable source for 3rd party fonts?",
```
Comment (mattscilipoti) [inline slips.json]: Not in https://github.com/ga-wdi-lessons/css-review

File: slips.json [hunk 4]
```diff
@@ -87,12 +98,12 @@ var data_slips = {
...
     "List at least 3 methods added to a class when it inherits from ActiveRecord::Base.",
-    "Describe what Sinatra is and what it is used for",
-    "Walk through the steps of the Sinatra MVC request-response cycle",
+    "Describe what Rails is and what it is used for",
+    "Walk through the steps of a MVC request-response cycle",
     "Explain what REST is and why we use it",
     "List the 5 HTTP request methods. How do they relate to the 4 crud actions?",
     "Whiteboard the HTML for a form that creates a new instance of a Student model.",
-    "What is the purpose of sinatra's layout.erb file?",
+    "What is the purpose of Rails' layout.erb file?",
```
Comment (mattscilipoti) [inline slips.json]: application.html.erb

**Comment by mattscilipoti** [comment]
I found a few questions that I don't think we have covered yet (adv. css, dbs).  Feel free to grab any questions from earlier weeks.

File: slips.json (hunk 1 above)
Comment (nolds9) [inline slips.json:43]: I think it means in the context of arrays and objects

**Pull request merged**